api/v1alpha1/runner_webhook.go

```diff
@@ -76,6 +76,16 @@ func (r *Runner) Validate() error {
 		errList = append(errList, field.Invalid(field.NewPath("spec", "repository"), r.Spec.Repository, err.Error()))
 	}
 
+	err = r.Spec.ValidateWorkVolumeClaimTemplate()
```

Comment:
Thanks for adopting the existing pattern of `ValidateFoo` here!

This made me realize that we might want to introduce another centralized function like `RunnerSpec.Validate()` that calls out to the three `Validate*` functions in the future, so that it's a bit more maintainable(in terms of less chance to accidentally missed adding another new `ValidateFoo` call to one out of three places)

// Just take this as a pure comment, not a request for change. It's out of the scope of this pull request. I'd greatly appreciate it if you could submit another pr for that though!

Reply:
That makes a lot of sense, I'll follow up with a separate pr for that if that works for you!